com.unity.sg2/Editor/GraphDeltaRegistry/Registry/MultiplyNode.cs

```diff
@@ -0,0 +1,114 @@
+using UnityEditor.ShaderFoundry;
+using UnityEditor.ShaderGraph.GraphDelta;
+using System.Linq;
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph.GraphDelta
+{
+
+    internal class MultiplyNode : INodeDefinitionBuilder
+    {
+        RegistryFlags IRegistryEntry.GetRegistryFlags() => RegistryFlags.Func;
+        RegistryKey IRegistryEntry.GetRegistryKey() => new RegistryKey { Name = "Multiply", Version = 1 };
+
+        #region Names
+        public readonly static string kInputA = "A";
+        public readonly static string kInputB = "B";
+        public readonly static string kOutput = "Output";
+        #endregion
+
+        #region Calc
+        private static int truncate(int a, int b) // ignore scalars for truncation.
+            => a == 1 ? b : b == 1 ? a : Mathf.Min(a, b);
+        private static int resolve(int dim, int resolved) // a scalar dimension ignores the resolved type.
+            => dim == 1 ? 1 : Mathf.Min(dim, resolved);
+
+        private static void GetDim(FieldHandler field, out int length, out int height)
+        {
+            height = (int)GraphTypeHelpers.GetHeight(field);
+            length = (int)GraphTypeHelpers.GetLength(field);
+        }
+
+        private static bool calcResolve(NodeHandler node, out int length, out int height)
+        {
+            length = 1;
+            height = 1;
+
+            var inputPorts = node.GetPorts().Where(e => e.IsInput);
+            var connectedFields = inputPorts.Select(e => e.GetConnectedPorts().FirstOrDefault()?.GetTypeField()).Where(e => e != null);
+            bool hasVector = inputPorts.Count() != connectedFields.Count();
+
+            foreach (var field in connectedFields)
+            {
+                // TODO: resolve primitive/precision.
+                GetDim(field, out var fieldLength, out var fieldHeight);
+                length = truncate(length, fieldLength);
+                height = truncate(truncate(length, height), fieldHeight);
+                hasVector |= fieldLength > 1 && fieldHeight == 1;
+            }
+            return hasVector;
+        }
```

Comment:
The above is just for the special type resolution rules. I'll be refactoring dynamics and this in a follow-up so that they follow more similar rules.

Multiply still has a special case where unconnected nodes only take the length of the provoking connection.